rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,127 @@
+---
+feature: meson_builds_nix
+start-date: 2022-08-25
+author: Anderson Torres
+co-authors: @p01arst0rm
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Use meson as an alternative build system for the reference implementation of Nix.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the reference implementation of Nix evaluator, as well as its companion toolset generated from the Nix source tree, is built using the typical `./configure` shell script that relies on autoconf, along with standard GNU Make utility.
+
+Over time, this build system has been modified to keep up with the development needs of Nix projec. However, it has reached a state where the build system became clunky and plastered, hard to understand and modify, consequently making improvements to the project as a whole very difficult.
+
+In particular, many changes have been introduced that impact compatibility outside Linux and NixOS niches. These issues can hinder development on other platforms, including but not limited to Unix-like systems.
+
+In light of this state of things, propose a novel alternative to the current buildsystem infrastructure.
+
+We expect to accomplish, among other goals,
+
+- better code structuring and documentation;
+- improved cross-platform support, especially in other programming environments, including but not limited to Unix-like operating systems;
+- shorter build times;
+- improved unit testing;
+- an overall improved user experience.
+
+# Detailed design
+[design]: #detailed-design
+
+A carefully crafted set of Meson files should be included in the Nix repository, in order to describe how to deploy the Nix repository, generating all the expected artifacts (command line tools, libraries, configuration files etc.)
+
+This novel build infrastructure should be able to provide at least feature parity with the current  quasi-autotools implementation, albeit with a different user interface.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Currently @p01arst0rm is writing an implementation from scratch.
+
+Some typical expected interactions are:
+
+- Configuring: `meson setup build_dir`
+- Building: `ninja -C build_dir` (or the backend-agnostic `meson -C build_dir compile`)
+- Installing: `ninja -C build_dir install` (or the backend-agnostic `meson -C build_dir install`)
```

Comment:
Here or somewhere else, it could be useful to compare the bulid steps with Meson to the current ones. So that people could get a more intuitive idea of the improvements.

Reply:
the issue is that the current build steps are not universal for all "supported" platforms. the main improvement is simply having a build system that works on all platforms (even if the compiled binary has issues etc)

Reply:
I would state this more clearly in this RFC too.

Reply:
it is, the RFC states that the buildsystem allows multi platform support which will allow developers to more easily fix issues with nix on darwin, frebsd, reactos, haiku, etc. the idea is to increase development progress, decrease build complexity & multi platform support. @AndersonTorres maybe this needs to be highlighted in more detail?

Reply:
@p01arst0rm I rewrote the RFC. I believe the essence stays the same.